### PR TITLE
Add Architecture and design doc

### PR DESCRIPTION
**Pull request by yokawasa**
# Description

- Add Architecture and design https://github.com/st-tech/gatling-operator/pull/40/commits/7e94cfa2f16f948b6b360be6f599363232fa0185
- Add "Debug and Trace Gatling Load Testing" section in user guide https://github.com/st-tech/gatling-operator/pull/40/commits/ecb70b2a242c3f132be7303040142e02acdc89d5
- Add note(caution) on ConfigMap limitation to user guide https://github.com/st-tech/gatling-operator/pull/40/commits/18fff1ff8a917857a5c7c4220f28519fb817cc2a

relevant issue 
- https://github.com/st-tech/gatling-operator/issues/22
- https://github.com/st-tech/gatling-operator/issues/12

### CODE REVIEW

**Review by tmrekk121 — APPROVED** [review]
lgtm!

**Review by yokochin0105 — APPROVED** [review]
lgtm

File: docs/user-guide.md [hunk 1]
```diff
@@ -198,6 +199,41 @@ spec:
...
 For a full sample manifest, please check [this](../config/samples/gatling-operator_v1alpha1_gatling02.yaml).
 
+📝 **Caution**: Please be noted that the data stored in a ConfigMap cannot exceed 1 MiB (ref [this](https://kubernetes.io/docs/concepts/configuration/configmap/)). If you need to store files that are larger than this limit, you may want to consider create Custom Gatling Image to bundle them in the container.
+
+### Debug and Trace Gatling Load Testing
+
+As you can see in the section of [Create Custom Gatling Image to bundle Gatling Load Testing Files](#create-custom-gatling-image-to-bundle-gatling-load-testing-files), you can check the logging output of each Gatling load testing via container log. But if you want to know more details on what's going on in Gatling load testing, you can leverage `logback.xml`.
+You can debug Gatling with `logback.xml` which is supposed to be located in the Gatling conf directory (see [gatling/conf/logback.xml](https://github.com/st-tech/gatling-operator/blob/main/gatling/conf/logback.xml)).
+
+For example, here is default logback configuration which allows to print debuggin information to the console.
```
Comment (kayamin) [inline docs/user-guide.md:209]:
```suggestion
For example, here is default logback configuration which allows to print debugging information to the console.
```

File: docs/architecture.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+- [Gatling Operator Architecture and Design](#gatling-operator-architecture-and-design)
+	- [Architecture Overview](#architecture-overview)
+	- [Gatling Runner Pod as Multi Container Pod](#gatling-runner-pod-as-multi-container-pod)
+
+<!-- /TOC -->
+
+## Architecture Overview
+
+The diagram below shows how Gatling Operator works in conjunction with Gatling Custom Resource (CR).
+Based on Gatling CR, actions in the lifecycle of distributed Gatling load testing such as running load testing, generating reports, sending notification message, and cleaning up the resources are executed by relevant custom controller (`Gatling Controller` in figure below).
```
Comment (kayamin) [inline docs/architecture.md:14]: [IMO] I think using brackets for long examples make sentence easier to understand.

```suggestion
Based on Gatling CR, actions in the lifecycle of distributed Gatling load testing (such as running load testing, generating reports, sending notification message, and cleaning up the resources) are executed by relevant custom controller (`Gatling Controller` in figure below).
```

**Review by kayamin — APPROVED** [review]
Thank you for updating documents. I commented some nits but as a whole it looks good to me 👍

File: docs/architecture.md (hunk 2 above)
Comment (yokawasa) [reply docs/architecture.md:14]: @kayamin Thanks for the suggestion. That makes sense!

**Comment by yokawasa** [comment]
@kayamin @tmrekk121 @yokochin0105 thanks for the view! I'm merging this

**Pull request merged**